### PR TITLE
config: multiple values in RemoteConfig (URLs and Fetch)

### PR DESCRIPTION
**Pull request by smola**
* Change `URL string` to `URL []string` in `RemoteConfig`, since
  git allows multiple URLs per remote. See:
  http://marc.info/?l=git&m=116231242118202&w=2

* Fix marshalling of multiple fetch refspecs.

### CODE REVIEW

**Comment by smola** [comment]
This will be useful in `src-d/borges` to store all endpoints for each repository. It will be required to implement push to multiple URLs (https://github.com/src-d/go-git/issues/500) at some point too.

**Comment by bzz** [comment]
Looks like rebase might be in order.

File: remote.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ func (r *Remote) Config() *config.RemoteConfig {
 }
 
 func (r *Remote) String() string {
-	fetch := r.c.URL
-	push := r.c.URL
+	fetch := r.c.URLs[0]
```
Comment (ajnavarro) [inline remote.go]: this can throw panics

File: remote_test.go [hunk 2]
```diff
@@ -27,33 +27,33 @@ type RemoteSuite struct {
 var _ = Suite(&RemoteSuite{})
 
 func (s *RemoteSuite) TestFetchInvalidEndpoint(c *C) {
-	r := newRemote(nil, &config.RemoteConfig{Name: "foo", URL: "http://\\"})
+	r := newRemote(nil, &config.RemoteConfig{Name: "foo", URLs: []string{"http://\\"}})
```
Comment (ajnavarro) [inline remote_test.go]: Add a test with a RemoteConfig with a nil slice of URLs. It will panic
